### PR TITLE
#349 uniqe class name

### PR DESCRIPTION
**Pull request by taltal115**

### CODE REVIEW

File: src/modules/persistence/class.persistence.service.ts [hunk 1]
```diff
@@ -8,9 +8,13 @@ import { ClassDbModel } from 'models/class.db.model';
...
     private collection: Collection<ClassDbModel>;
     private logger = new Logger('ClassPersistenceService');
-    constructor(private dbService: DbService, private schedulePersistenceService: SchedulePersistenceService) {
+    constructor(
+        private dbService: DbService,
+        private schedulePersistenceService: SchedulePersistenceService,
+    ) {
         const db = this.dbService.getConnection();
         this.collection = db.collection<ClassDbModel>('classes');
+        this.collection.createIndex( { name: 1 }, {unique: true} );
```
Comment (dimkinv) [inline src/modules/persistence/class.persistence.service.ts]: this happens everytime server is starting up. there are a lot of open questions here:
would it fail? 
would it create multiple indexes?
what would happen if multiple server instances will create index at the same time?

I would recommend to search the db with `findOne` to check uniqueness. 
This was you will also be able to test this code.

Comment (taltal115) [reply src/modules/persistence/class.persistence.service.ts]: Creating indexes in MongoDB is an idempotent operation. So running db.names.createIndex({name:1}) would create the index only if it didn't already exist.

The deprecated (as of MongoDB 3.0) alias for createIndex() is ensureIndex() which is a bit clearer on what createIndex() actually does.

Would it if i put out this functionality to a function?
what other way you suggest creating an index?
i know in mongoose you use a schema for a collection and you set there the index.

Comment (dimkinv) [reply src/modules/persistence/class.persistence.service.ts]: Ok. I would still put the logic in code and check for uniqness myself. This way we would be able to test this logic and not rely on the database to do it.
Though youre adding the index from code so as long as you will lock this line in unittests looks good for me.

Comment (taltal115) [reply src/modules/persistence/class.persistence.service.ts]: ok. ill be on it tomorrow.

Comment (taltal115) [reply src/modules/persistence/class.persistence.service.ts]: yes its done

**Pull request merged**